designcompose/src/main/java/com/android/designcompose/Shape.kt

```diff
@@ -0,0 +1,620 @@
+package com.android.designcompose
+
+import android.graphics.PointF
+import androidx.compose.ui.geometry.CornerRadius
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.geometry.RoundRect
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.asAndroidPath
+import androidx.compose.ui.graphics.asComposePath
+import androidx.core.graphics.minus
+import androidx.core.graphics.plus
+import com.android.designcompose.serdegen.BoxShadow
+import com.android.designcompose.serdegen.Dimension
+import com.android.designcompose.serdegen.StrokeAlign
+import com.android.designcompose.serdegen.StrokeCap
+import com.android.designcompose.serdegen.ViewShape
+import com.android.designcompose.serdegen.ViewStyle
+import java.util.Optional
+
+/// ShapePaths is a set of paths derived from a shape and style definition. These
+/// paths are based on the style information (known at document conversion time)
+/// and layout information (only known at runtime).
+///
+/// In the general path case, the various fills, strokes, clips, and shadow bounds
+/// are computed using "Path Operations" (boolean ops on paths, and offset paths
+/// computed using Skia's "getFillPath" function).
+///
+/// In some special cases, like frames and rectangles, a simpler algorithm is used
+/// because the geometry is trivial to calculate. This results in simpler paths to
+/// render, and may be less compute intensive.
+class ShapePaths(
+    /// The paths that need to be filled with the Background brush for
+    /// this shape.
+    val fills: List<Path>,
+
+    /// The paths that need to be filled with the Stroke brush for this
+    /// shape.
+    val strokes: List<Path>,
+
+    /// The paths that need to be clipped out (for outset shadows that
+    /// don't paint under the background) or clipped in for inset shadows.
+    ///
+    /// These paths are different from the "fills" because the shadow clip
+    /// is based on the stroke bounds, and not the background bounds.
+    val shadowClips: List<Path>,
+
+    /// The paths that are filled for shadows. The shadow style field is
+    /// referenced, which includes details like the inset/outset of the
+    /// shadow, its offset, and its color.
+    val shadowFills: List<ComputedShadow>,
+)
+
+class ComputedShadow(val fills: List<Path>, val shadowStyle: BoxShadow)
```

Comment:
Data class?
```suggestion
data class ComputedShadow(val fills: List<Path>, val shadowStyle: BoxShadow)
```

Reply:
Yeah, OK. I always forget that `data class` exists. We don't compare these or dump them to string anywhere, but I'll see if it applies (to this and ShapePaths).

Reply:
Ok, I didn't do this in this change because we don't compare these and not all of the contents are directly comparable in that way (but will try to remember `data class` for new objects that need comparisons & eqality).